File: src/main/java/art/arcane/racbot/config/CommandRegistrationMode.java
package art.arcane.racbot.config;

public enum CommandRegistrationMode {
  GLOBAL,
  GUILD;

  public static CommandRegistrationMode fromValue(String value) {
    if (value == null || value.isBlank()) {
      return GLOBAL;
    }
    try {
      return CommandRegistrationMode.valueOf(value.trim().toUpperCase());
    } catch (IllegalArgumentException ex) {
      return GLOBAL;
    }
  }
}

File: lib/src/config/command_registration_mode.dart
enum CommandRegistrationMode {
  global,
  guild;

  static CommandRegistrationMode fromValue(String? value) {
    String normalized = value == null ? '' : value.trim().toUpperCase();
    return switch (normalized) {
      'GUILD' => CommandRegistrationMode.guild,
      _ => CommandRegistrationMode.global,
    };
  }
}
